packages/transactions/lib/transactions.js

```diff
@@ -0,0 +1,208 @@
+import {ApiRpc} from '../client/api-rpc';
+import uuid from 'uuid';
+import {getRealm} from '../core/realm';
+import {DOCK_TOKEN_UNIT} from '../core/format-utils';
+import {fetchTransactions} from '../core/subscan';
+import BigNumber from 'bignumber.js';
+import {Accounts} from './accounts';
+import { NetworkManager } from './network-manager';
+
+export const TransactionStatus = {
+  InProgress: 'pending',
+  Failed: 'failed',
+  Complete: 'complete',
+};
+
+/**
+ * Transactions Module
+ */
+export class Transactions {
+  constructor() {
+    this.transactions = [];
+  }
+
+  /**
+   * Get transactions module instance
+   * @returns module instance
+   */
+  static getInstance() {
+    if (!Transactions.instance) {
+      Transactions.instance = new Transactions();
+    }
+
+    return Transactions.instance;
+  }
+  
+  /**
+   * Load external transactions for the given account
+   * 
+   * @param {string} account 
+   */
+  async loadExternalTransactions(account) {
+    const realm = getRealm();
+    const dbTransactions = realm.objects('Transaction').toJSON();
+  
+    const handleTransaction = tx => {
+      if (tx.from !== account && tx.to !== account) {
+        return;
+      }
+  
+      if (dbTransactions.find(item => item.hash === tx.hash)) {
+        return;
+      }
+  
+      const newTx = {
+        amount: BigNumber(tx.amount).times(DOCK_TOKEN_UNIT).toString(),
+        feeAmount: tx.fee,
+        recipientAddress: tx.to,
+        fromAddress: tx.from,
+        id: tx.hash,
+        hash: tx.hash,
+        network: 'mainnet',
+        status: 'complete',
+        date: new Date(parseInt(tx.block_timestamp + '000', 10)),
+      };
+  
+      realm.write(() => {
+        realm.create('Transaction', newTx, 'modified');
+      });
+    };
+  
+    let data;
+    let page = 0;
+  
+    do {
+      data = await fetchTransactions({address: account, page});
+      data.transfers.forEach(handleTransaction);
+      page++;
+    } while (data.hasNextPage);
+  }
+
+  /**
+   * Load transactions for the current accounts
+   * 
+   * @returns transactions
+   */
+  async loadTransactions() {
```

Comment:
I have concerns about how much data we're going to end up caching in the realm db if we constantly load all txns for all accounts. Have you thought about how we manage that?

Reply:
yeah i have the same concern. There is a task in the doc yeah i have the same concern. https://www.notion.so/dockteam/Wallet-sdk-code-review-25d2f2d7644c470ea5a1012a8cd7c871

It would be better to just fetch on demand, and also add pagination to it. I assume the app will just fetch from network when the user opens the account details page. And we will not cache anything in advance.

Also, with pagination, the user will be able to load older transactions, and we can keep caching this for future usage.

Reply:
we probably don't need to cache too much between sessions either - especially old stuff